resources/file_input.rb

```diff
@@ -24,23 +24,37 @@
 property :template_source, String, default: 'file-input.conf.erb'
 
 action :create do
-  log_name = new_resource.name
-  template "/etc/rsyslog.d/#{new_resource.priority}-#{new_resource.name}.conf" do
-    mode '0664'
-    owner node['rsyslog']['user']
-    group node['rsyslog']['group']
-    source new_resource.template_source
-    cookbook new_resource.cookbook_source
-    variables 'file_name' => new_resource.file,
-              'tag' => log_name,
-              'state_file' => log_name,
-              'severity' => new_resource.severity,
-              'facility' => new_resource.facility
-    notifies :restart, "service[#{node['rsyslog']['service_name']}]", :delayed
+  create_template(:create)
+end
+
+action :delete do
+  create_template(:delete)
+end
+
+action_class do
+  def log_name
+    new_resource.name
   end
 
-  service node['rsyslog']['service_name'] do
-    supports restart: true, status: true
-    action [:enable, :start]
+  def create_template(create_action)
+    template "/etc/rsyslog.d/#{new_resource.priority}-#{new_resource.name}.conf" do
+      mode '0664'
+      owner node['rsyslog']['user']
+      group node['rsyslog']['group']
```

Comment:
Hey @bish0polis 

Taking another look at this these would be much better as properties on the resource which default to these values for now.

Thanks